src/index.ts

```diff
@@ -1,7 +1,79 @@
+import { euclidean } from 'ml-distance-euclidean';
 /**
- * My module
- * @returns A very important number
+ * DBSCAN
+ * @returns list of cluster，-1 is noise
  */
-export function myModule(): number {
-  return 42;
+interface IPoint {
+  index: number;
+  value: number[];
+  label: number;
+}
+function rangeQuery(
+  cur: IPoint,
+  data: IPoint[],
+  epsilon: number,
+  distance: (p: number[], q: number[]) => number,
+) {
+  return data.filter((point) => distance(point.value, cur.value) <= epsilon);
+}
+export function dbscan(
+  points: number[][],
+  epsilon: number = 1,
+  minPoints: number = 2,
+  distance: (p: number[], q: number[]) => number = euclidean,
+) {
+  if (!(points instanceof Array)) {
```

Comment:
When methods have optional parameters we like to put an 'options' object with default value '{}' and then to deconstruct it.

Like in this example https://github.com/mljs/spectra-processing/blob/953886481787535ece7ddcd6d60d9301ef61ec6a/src/xy/xyFilterX.ts#L5-L33

Reply:
ok